docs/modules/ROOT/pages/guardrails.adoc

```diff
@@ -0,0 +1,236 @@
+= Guardrails - Controlling the Chaos
+
+Guardrails are mechanisms that let you validate the output of the LLM to ensure it meets your expectations.
+Typically, you can:
+
+- Ensure that the format is correct (e.g., it is a JSON document with the right schema)
+- Detect hallucinations (especially in the context of a RAG)
+
+NOTE: At the moment, Quarkus Langchain4J only offers output guardrails.
+Input guardrails are not yet implemented but are planned.
+
+image::guardrails.png[width=600,align="center"]
+
+== Implementing Output Guardrails
+
+Output guardrails are _functions_ invoked once the LLM has produced its output.
+They are implemented as CDI beans and must implement the `io.quarkiverse.langchain4j.guardrails.OutputGuardrail` interface.
+
+Output guardrails can receive:
+
+- The output of the LLM
+- The output of the LLM, the current memory, and the augmentation results (the data chunks that were added to the prompt in the context of a RAG)
```

Comment:
Is the first line meant to be something else?

Reply:
there are 2 validate methods, one simply taking only the response of the LLM and another one taking a record for more context.

Let me rephrase (actually there is a section about that below)